### PR TITLE
Add open in safe mode

### PR DESCRIPTION
**Pull request by srvkmr130**

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -208,7 +208,10 @@ $(document).ready(function () {
                    searchYoutube(data.result.parameters.any);
           // chrome.tabs.create({ 'url': 'https://www.youtube.com/results?search_query=' + data.result.parameters.any });
                 } else if (data.result.metadata.intentName === "open") {
-                   chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website });
+                   chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website })
```
Comment (gauthamzz) [inline background.js]: you missed a ; at the end

File: background.js [hunk 2]
```diff
@@ -208,7 +208,10 @@ $(document).ready(function () {
...
+                } else if (data.result.metadata.intentName === "incognito"||data.result.metadata.intentName === "safemode"||data.result.metadata.intentName === "privatebrowsing"||data.result.metadata.intentName === "openinsafemode") {
```
Comment (gauthamzz) [inline background.js]: intentName is only incognito. no need of the second one

File: background.js [hunk 3]
```diff
@@ -208,7 +208,10 @@ $(document).ready(function () {
...
+                   chrome.windows.create({url:"http://www.google.com",incognito: true});
+                   Speech("Have   you   allowed  me  to   open   in   incognito  mode ?    If not   go  to chrome extension and give the permission");
```
Comment (gauthamzz) [inline background.js]: No need for this line

File: background.js [hunk 4]
```diff
@@ -208,7 +208,10 @@ $(document).ready(function () {
                    searchYoutube(data.result.parameters.any);
           // chrome.tabs.create({ 'url': 'https://www.youtube.com/results?search_query=' + data.result.parameters.any });
                 } else if (data.result.metadata.intentName === "open") {
-                   chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website });
+                   chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website })
+                } else if (data.result.metadata.intentName === "incognito"||data.result.metadata.intentName === "safemode"||data.result.metadata.intentName === "privatebrowsing"||data.result.metadata.intentName === "openinsafemode") {
+                   chrome.windows.create({url:"http://www.google.com",incognito: true});
```
Comment (gauthamzz) [inline background.js]: Looks right

File: background.js [hunk 5]
```diff
@@ -219,7 +222,7 @@ $(document).ready(function () {
                    tweet(data.result.parameters.any);
                 // chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website });
                 } else if (data.result.metadata.intentName === "maps") {
-		   chrome.tabs.create({ 'url': "https://www.google.com/maps/dir/" + data.result.parameters["geo-city"][0] + "/" + data.result.parameters["geo-city"][1] });
+		               chrome.tabs.create({ 'url': "https://www.google.com/maps/dir/" + data.result.parameters["geo-city"][0] + "/" + data.result.parameters["geo-city"][1] });
```
Comment (gauthamzz) [inline background.js:229]: Its too much indentation change :smile:

**Review by gauthamzz — CHANGES_REQUESTED** [review]

File: background.js [hunk 6]
```diff
@@ -209,6 +209,9 @@ $(document).ready(function () {
           // chrome.tabs.create({ 'url': 'https://www.youtube.com/results?search_query=' + data.result.parameters.any });
                 } else if (data.result.metadata.intentName === "open") {
                    chrome.tabs.create({ 'url': "http://www." + data.result.parameters.website });
+                } else if (data.result.metadata.intentName === "incognito"||data.result.metadata.intentName === "safemode"||data.result.metadata.intentName === "privatebrowsing"||data.result.metadata.intentName === "openinsafemode") {
+                   chrome.windows.create({url:"http://www.google.com",incognito: true});
+                   Speech("Have   you   allowed  me  to   open   in   incognito  mode ?    If not   go  to chrome extension and give the permission");
```
Comment (gauthamzz) [inline background.js]: Should this be only said when it doesnt open in incognito. can you make like that .everything else looks good

**Review by gauthamzz — CHANGES_REQUESTED** [review]

File: background.js (hunk 6 above)
Comment (srvkmr130) [reply background.js]: speech () was added to inform the user he /she must allow the permision to anna extension to allow in incognito..
So I thought it is a must line..
Should I remove it ?

Comment (gauthamzz) [reply background.js]: I think that speech should only come ,if they dont allow incognito

**Pull request closed without merging**